### ISSUE TITLE
[CAIA A11y Research - PILOT]: Appointments/Check-In, Community Care Self-Scheduling - ASYNC PILOT (June 2024)

### ISSUE THREAD

**Issue by sara-amanda** [department-of-veterans-affairs/va.gov-team#86740]
> [!IMPORTANT]
> - **TICKET STATUS:** `Review Completed`
>     - Christina Gednalske has requested a pilot for this project. 
> - **Team Contacts:** Christina OOO 6/26-7/5; contact Ben Brasso and/or Ciera Maddox
> - **CAIA ASYNC PILOT DATE:** 6/25/2024  `Feedback before July 8` Research Starts 7/18/2024 
> - **TYPES OF ASSISTIVE TECH:** 
>     - _Mobile device_ (only SK and Sara have mobile devices)
>     - _Screen reader_

# CAIA A11y Support
This is a child ticket of the research ticket [86183](https://github.com/department-of-veterans-affairs/va.gov-team/issues/86183) - created to track work, completed from this pilot session.
> [!TIP]
> **DEFINITION OF DONE**: Refer to the task list in this ticket, for actionable items.

## CAIA Assignee(s)

<details><summary>Toggle to view CAIA info</summary>

- @sara-amanda 
- @SarahKay8 

</details>

## Team Info

<details><summary>Toggle to view team info</summary>

- Team name: Unified Appointments Experience (Appointments and Check-in)
- OCTO-DE product owner: Kay Lawyer and Kristen McConnell
- Product manager: Jeff Roof, Ben Brasso
- Designer: Ciera Maddox, Christina Gednalske, Ben Brasso
- FE engineer: Simi Adebowale, Lee Delarm
- Product/team Slack channel: #appointments-cc-direct-scheduling (https://dsva.slack.com/archives/C06SQ48B205)

</details>

## Description

<details><summary>Toggle to view the description</summary>

**A CAIA a11y will participate as a user in a pilot of this research study.**

- **Researcher**: The VFS team researcher will test their prototype and conversation guide by conducting the pilot session with a CAIA A11Y.
- **User**: The CAIA A11Y team member will act as a real user in the pilot session.
- **Assistive Tech**: The CAIA A11Y team member will follow the prompts given by the VFS Researcher, while using and demonstrating how the behavior of a `screen reader` would behave.
- **Timing**: Feedback needed no later than July 8, 2024. Completed 6/25/2024

</details>

## Pilot Session

> [!CAUTION]
> **Specifically relating to the "Date and Time" page:**
> - This is not something they can accurately present in CodePen without a ton of engineering work
>     - [Date time](https://codepen.io/team/va-cie/pen/ZENbQvq) is what they are using for screen reader users - to convey selecting date and then time.  
>     - [Date time #2](https://codepen.io/team/va-cie/pen/bGyvgma) will be used for sighted users as a image standing, that they can talk through.
> - **Source**: [DSVA Slack via Christina on 6/24/2024](https://dsva.slack.com/archives/C8E985R32/p1719243932777049?thread_ts=1718670186.998409&cid=C8E985R32)

### List of Screens

📄 📱[CodePen collection](https://codepen.io/collection/GoLwaz) with the list of screens they are using.

### Supporting Artifacts

<details><summary>Toggle to view the pilot supporting artifacts</summary>

- [CodePen Prototype](https://codepen.io/team/va-cie/live/WNBQNxb)
- [Research plan](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/health-care/appointments/va-online-scheduling/initiatives/community-care-direct-scheduling/research/2024-06%20Community%20Care%20Self-Scheduling%20-%20Usability%20and%20Concept%20Testing/research-plan.md)
- [Research conversation guide](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/health-care/appointments/va-online-scheduling/initiatives/community-care-direct-scheduling/research/2024-06%20Community%20Care%20Self-Scheduling%20-%20Usability%20and%20Concept%20Testing/conversation-guide.md)
- [CodePen collection](https://codepen.io/collection/GoLwaz) with the list of screens they are using.

</details>

### Assistive Tech Options

> [!TIP]
> **DEVICE SUPPORT DURING RESEARCH SESSIONS**
> - **Available AT**: `Screen readers` on a `desktop` device is currently the primary source of AT that CAIA A11ys are able to support, in addition to `screen magnification` for user research sessions.
> - **Additional Specifications:** If there is a particular `test user`, `browser` or `operating system` you would like us to use, please let us know in advance via this ticket's comments, tagging the assignee - or reach out to @coforma-terry in the CAIA Slack Channel: [#sitewide-content-accessibility-ia ](https://dsva.slack.com/archives/C01K37HRUAH)

## Definition of Done

```[tasklist]
### A11y Tasks
- [x] **Request CAIA a11y**
- [x] **Share link to [Assistive Tech Pilot Guide](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/teams/CAIA/accessibility/assistive-tech-pilot-guide.md) with team, if they haven't reviewed it yet**
- [x] **Determine date and time for pilot:**  [Async]
- [x] `**NA - due to Async**` Update CAIA A11y Capacity Calendar: Add invite for pilot, when scheduled (Holds placed 5/22/2024)
- [x] Request AT device and tech to be used in sessions
- [x] [Receive AT device and tech from Team in research ticket](LINK) to be used in sessions to prep for pilot `Mobile and Screen Reader`
- [x] **Review research** [conversation guide](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/health-care/appointments/va-online-scheduling/initiatives/community-care-direct-scheduling/research/2024-06%20Community%20Care%20Self-Scheduling%20-%20Usability%20and%20Concept%20Testing/conversation-guide.md) and provide feedback in comments. `Feedback no later than 7/8/2024`
- [x] **Review prototype** in `CodePen` [CodePen Prototype](https://codepen.io/team/va-cie/live/WNBQNxb). [CodePen collection](https://codepen.io/collection/GoLwaz) with the list of screens they are using.  `See cautionary note in this ticket`
- [x] **Run pilot** for VFS Researcher on a **mobile device** with a **screen reader** using the [conversation guide](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/health-care/appointments/va-online-scheduling/initiatives/community-care-direct-scheduling/research/2024-06%20Community%20Care%20Self-Scheduling%20-%20Usability%20and%20Concept%20Testing/conversation-guide.md). `Feedback no later than 7/8/2024`
- [x] **Optional: Note Taking:** If there is only one a11y, notes cannot be taken due to pilot participation. Folder for Team in GH Deliverables for CAIA A11Ys. `In comments`
```

**Comment by SarahKay8** [comment]
## CodePen Audit - Environment: Talkback Android, Chrome 


### **Issue 1:** 
In the attached video, the current date format presents an issue where the day precedes the month, leading to confusion in screen reader outputs. For instance, instead of reading ‘Monday the 13th,’ VoiceOver and TalkBack announce it as ‘13-MON,’ which may not accurately convey the intended date information. This format also results in the focus indicator initially landing on the day digit alone (‘13’), causing a disjointed reading experience for users. We recommend adjusting the date format to ensure clear and coherent screen reader outputs, enhancing accessibility and usability for all users.



_**Recommendation**: Update headings to Mon the 13th and so on so forth_

https://github.com/department-of-veterans-affairs/va.gov-team/assets/102324990/66c55ee7-b3a7-48c8-9bb0-c060e06d7fdb

### **Issue 2:** 

When you have a “directions” link in your application that opens external mapping applications like Google Maps, it’s important to inform users, especially those with accessibility needs, about this behavior. Here’s how you can handle this effectively:

Providing Accessibility Information:

* Labeling the Link Clearly:
	•	Ensure the link text itself is descriptive. For instance, instead of just “Directions,” consider something like “Get directions on Google Maps.”
* Adding Context with ARIA Attributes:
	•	Use aria-label or aria-describedby attributes to provide additional context. For example:

<a href="ht[tps://maps.google.com/" ](https://maps.google.com/)aria-label="Link to get directions to Cheyenne Va Medical Center on Google Maps">Get directions on Google Maps</a>
![A2F989D2-CF62-42E2-BCCB-5099E81C25C0_4_5005_c](https://github.com/department-of-veterans-affairs/va.gov-team/assets/102324990/a4940759-7a19-4227-a4d2-fde4653052c7)


### **Issue 3:** 
After the “print” button on the main screen there is something hidden being read out as “optional description text for screenreaders” 



https://github.com/department-of-veterans-affairs/va.gov-team/assets/102324990/42e38bbc-a08a-49a2-9bbc-12fe0af0a0f3

### **Issue 4:** 

Appointments back button isn’t working on prototype have to start flow all over again to get to main screen
 

![8F1C894B-CE78-4F51-B547-E7A19FB6D026_4_5005_c](https://github.com/department-of-veterans-affairs/va.gov-team/assets/102324990/a6c0dc90-a4b1-408e-8512-34d6719464e0)

HI @benbrasso-agile6 - attached is my mobile audit for your codepen. None of these issues are launch blocking however I would consider these criticial. Let me know if additional info is needed. Best - SK! CC: @sara-amanda

**Comment by benbrasso-agile6** [comment]
Linking this ticket to our work space's ticket. Thanks!
